File: LivrusVirtus/Delphi7/CLX/dbExpress/fontes/ConProfissoes.pas
unit ConProfissoes;

interface

uses
  SysUtils, Types, Classes, Variants, QTypes, QGraphics, QControls, QForms, 
  QDialogs, QStdCtrls, Consultas, FMTBcd, Provider, DBClient, DB, SqlExpr,
  QGrids, QDBGrids, QExtCtrls;

type
  TfrmConProfissoes = class(TfrmConsultas)
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure btnLimparClick(Sender: TObject);
    procedure btnPesquisarClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmConProfissoes: TfrmConProfissoes;

implementation

uses Principal, ConsultasSQL;

{$R *.xfm}

procedure TfrmConProfissoes.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
frmPrincipal.Profissoes2.Enabled:=true;
inherited;
end;

procedure TfrmConProfissoes.btnLimparClick(Sender: TObject);
begin
  inherited;
LimparDados(ConsSQL.Profissao('S','0',''));
end;

procedure TfrmConProfissoes.btnPesquisarClick(Sender: TObject);
begin
  inherited;
PesquisarDados(ConsSQL.Profissao('S','', edtDescricao.Text),
'Profissão não encontrada!');
end;

procedure TfrmConProfissoes.FormShow(Sender: TObject);
begin
  inherited;
btnLimparClick(self);
end;

end.

File: LivrusVirtus/Kylix3/cpp/fontes/ConProfissoes.cpp
//---------------------------------------------------------------------------

#include <clx.h>
#pragma hdrstop

#include "ConProfissoes.h"
#include "ConsultasSQL.h"
#include "Principal.h"
//---------------------------------------------------------------------------
#pragma link "Consultas"
#pragma resource "*.xfm"
TfrmConProfissoes *frmConProfissoes;
//---------------------------------------------------------------------------
__fastcall TfrmConProfissoes::TfrmConProfissoes(TComponent* Owner)
        : TfrmConsultas(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmConProfissoes::FormClose(TObject *Sender,
      TCloseAction &Action)
{
frmPrincipal->Profissoes2->Enabled=true;
TfrmConsultas::FormClose(Sender, Action);
}
//---------------------------------------------------------------------------
void __fastcall TfrmConProfissoes::btnLimparClick(TObject *Sender)
{
LimparDados(ConsSQL->Profissao('S',"0",""));
}
//---------------------------------------------------------------------------
void __fastcall TfrmConProfissoes::btnPesquisarClick(TObject *Sender)
{
PesquisarDados(ConsSQL->Profissao('S',"", edtDescricao->Text.c_bstr()),
"Profissão não encontrada!");
}
//---------------------------------------------------------------------------
void __fastcall TfrmConProfissoes::FormShow(TObject *Sender)
{
btnLimparClick(this);
}
//---------------------------------------------------------------------------

